### PR TITLE
Identified fields in ui_advmode

### PR DESCRIPTION
**Pull request by AtomicChicken**

### CODE REVIEW

**Comment by dscorbett** [comment]
Many of these fields should be enums or bitfields instead of integers.

**Comment by AtomicChicken** [comment]
Kindly edit as you deem fit. I am inexperienced in this regard.

File: df.advmode.xml [hunk 1]
```diff
@@ -716,11 +718,11 @@
         </compound>
 
         <pointer name='unk_91'/>
-        <compound name='unk_v44_1' comment='assumed identity?'>
-            <int32_t/>
+        <compound name='assume_identity' comment='Manages the Assume Identity UI when the AssumeIdentity menu is open'>
+            <int32_t name='mode' comment='0 = SelectIdentity, 1 = CreateIdentity, 2 = SelectProfession, 3 = SelectWorship, 4 = SelectOrigin'/>
```
Comment (lethosor) [inline df.advmode.xml]: This should be an enum instead

**Comment by AtomicChicken** [comment]
Attempted to create the requested enum and added some ref-targets while I was at it. Might have messed up; please review.

**Comment by lethosor** [comment]
I should have been more clear, sorry. In this case, defining an inline enum (like [this](https://github.com/DFHack/df-structures/blob/caed82e28f6e9b9704eb10165a657ee912251e08/df.viewscreen.xml#L219-L222)) would be fine, since it's only used in one place (as far as I know, anyway). There's nothing wrong with a standalone enum, and I'm fine with merging that, but it's usually easier to use inline enums when they're only used once.
Unrelated: buildmaster failed to check out https://github.com/DFHack/df-structures/commit/57468eda0245a04127efb3d6b3a9b96dfcd60d0f for some reason, even though GitHub recognizes it and it doesn't look like you force-pushed or anything. I'm going to reopen this and see if that changes anything.

**Comment by lethosor** [comment]
Made a couple changes in eec1958, FYI

**Pull request merged**